controller/sharding/sharding.go

```diff
@@ -144,6 +160,11 @@ func RoundRobinDistributionFunction(db db.ArgoDB) DistributionFunction {
 	}
 }
 
+// NoShardingDistributionFunction returns a DistributionFunction that will process all shards
+func NoShardingDistributionFunction(shard int) DistributionFunction {
+	return func(c *v1alpha1.Cluster) int { return shard }
```

Comment:
As we are always assigning shard number 0 for No Sharding Algorithm, do we need to take shard as an input? We can directly return shard number as 0 here. WDYT?

Reply:
Agreed. I wanted to let the door open for a special scenario where a user would want to have sharding enabled but wants to set all the clusters processed by a specific shard. That would have required an additional parameter. So for now, let's just set it to a fixed value of 0.